reviews/views.py

```diff
@@ -77,6 +77,83 @@ def get_context_data(self, **kwargs):
         context['data_url'] = reverse("reviews:api:open", args=[self.committee])
 
         return context
+    
+
+class CommitteeMembersWorkloadView(GroupRequiredMixin, CommitteeMixin, generic.TemplateView):
+    #TODO: Look into improving the should_end_date variable. See notes
```

Comment:
This was a note to self. I've actually solved the problem I was running into. I'll just explain what was happening. So the old logic in the assign_reviewers function, looked like this:
```
if review.short_route and review.date_should_end is None:
        review.date_should_end = timezone.now() + \
            timezone.timedelta(
                weeks=settings.SHORT_ROUTE_WEEKS
            )
    elif review.date_should_end is not None:
        # We have no desired end date for long track reviews
        review.date_should_end = None
```
Besides the fact that I've implemented a new condition, for revisions, this code was causing a problem for me. Because there is the possibility that a review automatically gets a should_end_date in the start_assignment_phase util function, it would then revert to None, after manually assigning a route, even if I chose short route. With the new implementation, the should_end_date will always be updated according to the form instance value of the short_route on the assignment page.

It currently looks like this. ```route``` gets passed from the ReviewAssignView as ```route = form.instance.short_route```
```
    if review.proposal.is_revision:
        #Revisions should end in one week.
        review.date_should_end = timezone.now() + \
            timezone.timedelta(
                weeks=1
            )
    elif route == True:
        #The short route reviews should take two weeks.
        review.date_should_end = timezone.now() + \
            timezone.timedelta(
                weeks=settings.SHORT_ROUTE_WEEKS
            )
    elif route == False:
        # We have no desired end date for long track reviews
        review.date_should_end = None
```
As I am typing this, I am thinking that this makes the auto assignment of the should_end_date obsolete ... Might make sense to remove it, right?

Reply:
> As I am typing this, I am thinking that this makes the auto assignment of the should_end_date obsolete ... Might make sense to remove it, right?

It might? But there is an argument to be made for keeping it, as the secretary might be late doing the manual assign for some reason. If the automatic assign thinks it's fine for a short route, we might actually want the date_should_end to be set as well as that will prompt the app to start nagging the secretary if they are overdue.

In fact, we might even want to update the automatic assign to also take into account revision deadlines. 

It's an edgecase for sure, the secretary is very punctual, so it might not be the best argument for keeping it. (But their replacement might not be as punctual ;) )

Reply:
Ok, I've updated the logic to keep the auto generated date_should_end if it exists.